plugins/module_utils/ec2.py

```diff
@@ -315,10 +1204,419 @@ def normalize_ec2_vpc_dhcp_config(option_config):
     return config_data
 
 
-@AWSRetry.jittered_backoff(retries=10)
-def helper_describe_import_image_tasks(client, module, **params):
+def get_default_vpc(client, module: AnsibleAWSModule) -> Optional[Dict[str, Any]]:
     try:
-        paginator = client.get_paginator("describe_import_image_tasks")
-        return paginator.paginate(**params).build_full_result()["ImportImageTasks"]
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed to describe the import image")
+        vpcs = describe_vpcs(client, Filters=ansible_dict_to_boto3_filter_list({"isDefault": "true"}))
+    except AnsibleEC2Error as e:
+        module.fail_json_aws(e, msg="Could not describe default VPC")
+    return None if len(vpcs) == 0 else vpcs[0]
+
+
+def get_default_subnet(
+    client, module: AnsibleAWSModule, vpc: Dict[str, Any], availability_zone: Optional[str] = None
+) -> Optional[Dict[str, Any]]:
+    try:
+        subnets = describe_subnets(
+            client,
+            Filters=ansible_dict_to_boto3_filter_list(
+                {
+                    "vpc-id": vpc["VpcId"],
+                    "state": "available",
+                    "default-for-az": "true",
+                }
+            ),
+        )
+    except AnsibleEC2Error as e:
+        module.fail_json_aws(e, msg=f"Could not describe default subnets for VPC {vpc['VpcId']}")
+    if subnets:
+        if availability_zone is not None:
+            match = [subnet for subnet in subnets if subnet["AvailabilityZone"] == availability_zone]
+            if match:
+                return match[0]
+
+        # to have a deterministic sorting order, we sort by AZ so we'll always pick the `a` subnet first
+        # there can only be one default-for-az subnet per AZ, so the AZ key is always unique in this list
+        by_az = sorted(subnets, key=lambda s: s["AvailabilityZone"])
+        return by_az[0]
+    return None
+
+
+# EC2 instances modules utilities
+class EC2InstanceModule(AnsibleAWSModule):
```

Comment:
Please don't pull this out into module_utils.ec2

Doing so means that any additional parameters for ec2_instance will trigger a ridiculous number of integration tests